### PR TITLE
adds a workflow for macOS spotlight indexing

### PR DESCRIPTION
**Pull request by jpvajda**
## Discord username 

- `JV [John Vajda]#0767`

## Description of changes (updated or new workflows)

- adds a new directory macOS 
- adds a new workflow to enable and disable macOS spotlight indexing which can be a very CPU heavy process.

### CODE REVIEW

File: specs/macOS/spotlight-indexing.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
+name: Starts or stops the MacOS Spotlight indexing process mds_stores.
+command: |-
+    sudo mdutil -a -i {{switch}}
+tags:
+  - Apple
+  - MacOS
+  - mds_stores
+  - Spotlight
```
Comment (alokedesai) [inline specs/macOS/spotlight-indexing.yaml:9]: nit: Could we try to minimize the number of tags here? In general with workflows we've found fewer and more specific tags are better. Maybe Just `mds_stores` and `MacOS`?

**Comment by elviskahoro** [comment]
@jpvajda I'm just making a pass through our open PRs. Could you cull a couple of the tags for spotlight-indexing i.e. delete: the "Apple" and "MacOS" tags so that they're more specific

File: specs/macOS/spotlight-indexing.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+---
+name: Starts or stops the MacOS Spotlight indexing process mds_stores.
+command: |-
+    sudo mdutil -a -i {{switch}}
+tags:
+  - Apple
```
Comment (jpvajda) [inline specs/macOS/spotlight-indexing.yaml:6]:
```suggestion
```

File: specs/macOS/spotlight-indexing.yaml (hunk 1 above)
Comment (jpvajda) [inline specs/macOS/spotlight-indexing.yaml:9]:
```suggestion
```

**Comment by jpvajda** [comment]
@elviskahoro. No problem. I just remove those extra tags you suggested.

**Comment by elviskahoro** [comment]
thanks @jpvajda Are you in our Discord I can add the contributor role to your username? If you are please let me know what your username is, you can reply here or DM me in Discord

**Comment by jpvajda** [comment]
@elviskahoro  you can find me on Discord at `jpvajda` Thanks!

**Review by elviskahoro — APPROVED** [review]

**Pull request merged**